torchbenchmark/score/compute_score_v1.py

```diff
@@ -0,0 +1,193 @@
+"""
+Compute the benchmark score given a frozen score configuration and current benchmark data.
+"""
+import argparse
+import json
+import math
+import sys
+import os
+import re
+import yaml
+import importlib
+
+from enum import Enum
+from tabulate import tabulate
+from pathlib import Path
+from collections import defaultdict
+
+TORCHBENCH_V1_REF_DATA = Path(__file__).parent.joinpath("configs/v1/config-v1.yaml")
+
+def _get_model_task(model_name):
+    """
+    Helper function which extracts the task the model belongs to
+    by iterating over the Model attributes.
+    """
+    try:
+        module = importlib.import_module(f'torchbenchmark.models.{model_name}', package=__name__)
+    except:
+        raise ValueError(f"Unable to get task for model: {model_name}")
+    Model = getattr(module, 'Model')
+    return Model.task
+
+class TorchBenchScoreV1:
+    # ref_data: the YAML file or json file
+    def __init__(self, ref_data, spec_file, target):
+        if not ref_data:
+            ref_data = TORCHBENCH_V1_REF_DATA
+        self.norm = self._setup_benchmark_norms(ref_data)
+        self.norm_weights = self._setup_weights(self.norm)
+        # spec_file is not used in V1, this is just a placeholder
+        self.spec_file = spec_file
+        self.target = target
+        
+    def _filter_jit_tests(self, norm):
+        result_ref = dict()
+        result_ref['benchmarks'] = dict()
+        for jit_name in filter(lambda x: '-jit' in x, norm['benchmarks'].keys()):
+            left, sep, right = jit_name.rpartition('-jit')
+            eager_name = left + "-eager" + right
+            # We assume if a jit test exists, there must be an eager test
+            assert eager_name in norm['benchmarks'], f"Can't find eager test name {eager_test_name}"
+            result_ref['benchmarks'][jit_name] = dict()
+            result_ref['benchmarks'][jit_name]['jit_norm'] = norm['benchmarks'][jit_name]['norm']
+            result_ref['benchmarks'][jit_name]['eager_norm'] = norm['benchmarks'][eager_name]['norm']
+        return result_ref
+
+    def _test_to_config(self, name):
+        if "-freeze" in name:
+            name.replace("-freeze", "", 1)
+        test, model_name, device, mode = re.match(r"test_(.*)\[(.*)\-(.*)\-(.*)\]", name).groups()
+        return (test, model_name, device, mode)
+
+    def _config_to_weight(self, config):
+        # config weight rule in V1: 1x CPU Training, 2x GPU Training, 2x CPU Inference, 2x GPU Inference
+        test, model_name, device, mode = config
+        if test == "train" and device == "cpu":
+            return 1.0
+        else:
+            return 2.0
+        
+    # Generate the domain weights from the ref object
+    def _setup_weights(self, ref):
```

Comment:
The implementation of this is not easy to understand from the code. Could you include a brief explanation somewhere about how weights work for v1.0?

Reply:
Sure, will do.

Reply:
i agree, it seems like there is too much complexity here, and I wonder if it can be refactored so it is more readable and less lines of code.  

I guess it wouldn't change things that much to combine the 3 for loops, but it seems like you could do that.  

another possibility to consider is creating a data structure or two.  

Maybe one data-structure that wraps a reference_data file and exposes convenient APIs on it (e.g. to get particular numbers based on names or configs) so you don't have to do as much of the filtering and extracting in the TorchBenchScoreV1 class.

Maybe another (simpler?) data structure that replaces all the defaultdicts you construct directly here- Seems like what you're really constructing here is a 'weights' object, which probably could also include the 'norms'.  maybe could be called a 'scoremeta' object or something.  and have accessor methods that get the norm or weight for a particular benchmark name. 

this is a bigger change, so i'm not sure if it's critical to do now, but it would be nice to make this more readable- I doubt others could easily maintain or understand it currently.

Reply:
Very good point! I will create a data structure and some APIs to expose the numbers needed for weights.